tutorials/Python/neonUtilities/neonUtilitiesPython.md

```diff
@@ -0,0 +1,342 @@
+    45: Serbia [https]                   46: Spain (A Coruña) [https]       
+    47: Spain (Madrid) [https]           48: Sweden [https]                 
+    49: Switzerland [https]              50: Turkey (Denizli) [https]       
+    51: Turkey (Mersin) [https]          52: UK (Bristol) [https]           
+    53: UK (Cambridge) [https]           54: UK (London 1) [https]          
+    55: USA (CA 1) [https]               56: USA (IA) [https]               
+    57: USA (KS) [https]                 58: USA (MI 1) [https]             
+    59: USA (NY) [https]                 60: USA (OR) [https]               
+    61: USA (TN) [https]                 62: USA (TX 1) [https]             
+    63: Vietnam [https]                  64: (other mirrors)                
+    
+    
+    
+    
+    
+    
+    Selection: 56
+    
+    
+    
+    The downloaded binary packages are in
+    	/var/folders/_k/gbjn452j1h3fk7880d5ppkx1_9xf6m/T//Rtmpp3bvLx/downloaded_packages
+    
+    
+    
+
+
+Now import the `devtools` package. This does need to be run every time you use the code; for those of you who are familiar with R, `importr` is roughly equivalent to the `library()` function in R.
+
+
+```python
+devtools = importr('devtools')
+```
+
+Using the `devtools` package, install the `neonUtilities` package. As with `devtools`, you can theoretically skip this step in the future, but since we update `neonUtilities` frequently, we recommend re-installing regularly.
+Then import the package.
+
+
+```python
+devtools.install_github('NEONScience/NEON-utilities/neonUtilities');
+neonUtils = importr('neonUtilities')
+```
+
+## Stack data files
+
+The function `stackByTable()` in `neonUtilities` merges the monthly, site-level files the <a href="http://data.neonscience.org/home" target="_blank">NEON Data Portal</a> provides. Start by downloading the dataset you're interested in from the Portal. It will download as a single zip file. Note the file path it's saved to and proceed.
+
+The data stacker package comes with a data file, table_types. The data file is needed for the package to work, and rpy2 doesn't load data by default. So we need to load it to the session and then pass it back to the R environment.
+
+First, load the data file:
+
+
+```python
+neonUdata = neonUtils.__rdata__
+table_types = neonUdata.fetch('table_types')['table_types']
+```
+
+Now, pass it back to the R environment:
+
+
+```python
+robjects.globalenv['table_types'] = table_types
+```
+
+Now run the `stackByTable()` function to stack the data. It requires two inputs: 
+1. The data product identifier (DPID) of the data you're stacking. DPIDs can be found in the <a href="http://data.neonscience.org/data-product-catalog" target="_blank">Data Product Catalog</a> and are in the form DP#.######.001
+2. A file path, the path to the zip file you downloaded from the NEON Data Portal.
+
+
+```python
+neonUtils.stackByTable('DP1.10100.001','/Downloads/NEON_isotope-soil-distrib-periodic.zip');
```

Comment:
I had to have a ~ in front of my file path.  Is that standard?  Or just variation by computer?